datastream-server-api/src/main/java/com/linkedin/datastream/server/FlushlessEventProducerHandler.java

```diff
@@ -73,7 +76,7 @@ public void clear(String source, int partition) {
    */
   public void send(DatastreamProducerRecord record, String source, int sourcePartition, T sourceCheckpoint, SendCallback callback) {
     SourcePartition sp = new SourcePartition(source, sourcePartition);
-    CallbackStatus status = _callbackStatusMap.computeIfAbsent(sp, d -> new CallbackStatus());
+    CallbackStatus status = _callbackStatusMap.computeIfAbsent(sp, d -> new CallbackStatusWithComparableOffsets());
```

Comment:
What's the purpose of the abstract `CallbackStatus` if there's only one implementation?